### PR TITLE
Apache common dogstream

### PR DESCRIPTION
**Pull request by abruehl**
An example dogstream parser for apache common and combined log format.  
It's rough but works just fine.  

Creates two metrics
apache.net.requests
apache.net.request_size

These metrics are tagged by verb (get, post, put, etc...) and status code family (2xx, 3xx, 4xx, etc)

### CODE REVIEW

File: dogstream/apache.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+both config lines assume your logs are in /var/log/httpd/access.log and this file is
+at /etc/dd-agent/dogstream/apache.py
+
+"""
+import time
+from datetime import datetime
+import re
+import math
+
```
Comment (irabinovitch) [inline dogstream/apache.py]: Imported but not used.

File: dogstream/apache.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    r'\[(?P<time>.+)\]',                # time %t
+    r'"(?P<verb>[A-Z]+)',               # HTTP Verb (GET, POST, etc)
+    r'(?P<request>.*)"',                # request "%r" (minus verb)
+    r'(?P<status>[0-9]+)',              # status %>s
+    r'(?P<size>\S+)',                   # size %b (careful, can be '-')
+    r'"(?P<referrer>.*)"',              # referrer "%{Referer}i"
+    r'"(?P<agent>.*)"'                  # user agent "%{User-agent}i"
+]
+
+COMMON_PATTERN_PARTS  = [
```
Comment (irabinovitch) [inline dogstream/apache.py]: E221 multiple spaces before operator

File: dogstream/apache.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        attr_name, attr_val = attr_pair.split('=')
+        attr_name = attr_name.strip()
+        attr_val = attr_val.strip()
+        attr_dict[attr_name] = attr_val
+
+    # Return the output as a tuple
+    return (metric_name, date, metric_value, attr_dict)
+
+def test():
+    import sys
```
Comment (irabinovitch) [inline dogstream/apache.py]: F401 'sys' imported but unused

**Comment by irabinovitch** [comment]
@abruehl  Thank you for the contribution and for being part of the Datadog community!  Our team will review and be in touch with any feedback. In the mean time, I've noted a few quick items that our tests/style checks caught.

**Comment by remh** [comment]
@abruehl Thanks. I'm moving this to a "To be determined" milestone until the comments made by @irabinovitch are addressed.

**Comment by irabinovitch** [comment]
@abruehl Were you by any chance able to review the comments and failing tests? We'd love to include this in an upcoming release.

**Comment by abruehl** [comment]
@irabinovitch I have not looked yet. I'll take a look this weekend.

**Comment by mikecouk** [comment]
We've just tried this with gunicorn combined logs and it seems to work !

**Comment by irabinovitch** [comment]
@mikecouk Awesome. Thanks for the feedback.  Happy to hear it was helpful to you.

@abruehl Let us know if you need any help troubleshooting the tests.

**Comment by irabinovitch** [comment]
@abruehl Just checking back in. Would you be able to address the tests here when you have a moment? thanks.

**Comment by abruehl** [comment]
@irabinovitch That should do it.  Sorry it took so long.  

Not sure what the network failure is.  It does not look related to my change.  Let me now if you feel otherwise.

**Comment by jeremy-lq** [comment]
Thanks for the contribution! As dogstreams is deprecated in Agent6, I'm going to close this. We apologize for the delay. Datadog Logs supports Apache logs out of the box. Check out https://docs.datadoghq.com/logs/ for more information.

**Pull request closed without merging**